wakuv2/config.go

```
@@ -30,30 +32,43 @@ import (
+	MinPeersForFilter        int              `toml:",omitempty"` // Indicates the minimum number of peers required for using Filter Protocol
+	LightClient              bool             `toml:",omitempty"` // Indicates if the node is a light client
+	WakuNodes                []string         `toml:",omitempty"`
+	Rendezvous               bool             `toml:",omitempty"`
+	DiscV5BootstrapNodes     []string         `toml:",omitempty"`
+	Nameserver               string           `toml:",omitempty"` // Optional nameserver to use for dns discovery
+	Resolver                 ethdisc.Resolver `toml:",omitempty"` // Optional resolver to use for dns discovery
+	EnableDiscV5             bool             `toml:",omitempty"` // Indicates whether discv5 is enabled or not
+	DiscoveryLimit           int              `toml:",omitempty"` // Indicates the number of nodes to discover with peer exchange client
+	AutoUpdate               bool             `toml:",omitempty"`
+	UDPPort                  int              `toml:",omitempty"`
+	EnableStore              bool             `toml:",omitempty"`
+	StoreCapacity            int              `toml:",omitempty"`
+	StoreSeconds             int              `toml:",omitempty"`
+	TelemetryServerURL       string           `toml:",omitempty"`
+	DefaultShardPubsubTopic  string           `toml:",omitempty"` // Pubsub topic to be used by default for messages that do not have a topic assigned (depending whether sharding is used or not)
+	UseShardAsDefaultTopic   bool             `toml:",omitempty"`
+	ClusterID                uint16           `toml:",omitempty"`
+	EnableConfirmations      bool             `toml:",omitempty"` // Enable sending message confirmations
+	SkipPublishToTopic       bool             `toml:",omitempty"` // Used in testing
+}
+
+func (c *Config) Validate() error {
+	if c.LightClient && (c.EnablePeerExchangeServer || c.EnableDiscV5 || !c.EnablePeerExchangeClient) {
```

From your logic, if LightClient is set, the other config is not makes any influence as it will fail the validation. Can we only have LightClient config if that's intended?

---

Keeping `EnablePeerExchangeServer` / `EnablePeerExchangeClient` would make implementation of `wakuv2.Waku` clear IMO , it will also support changing Config in status-go side internally if needed. This is why i only add a validation for warning log rather return an error. I made these changes according this [comment](https://github.com/status-im/status-go/pull/4665/files#r1495271941) and following. WDYT? @igor-sirotin @kaichaosun  @chaitanyaprem